src/storage/wal_replayer.cpp

```diff
@@ -418,14 +411,18 @@ void WALReplayer::replayDropPropertyRecord(const kuzu::storage::WALRecord& walRe
 
 void WALReplayer::replayAddPropertyRecord(const kuzu::storage::WALRecord& walRecord) {
     if (isCheckpoint) {
+        // See comments at `replayNodeTableRecord`.
+        auto catalogForCheckpointing = getCatalogForRecovery(DBFileType::WAL_VERSION);
         auto tableID = walRecord.addPropertyRecord.tableID;
         auto propertyID = walRecord.addPropertyRecord.propertyID;
+        auto tableSchema = catalogForCheckpointing->getReadOnlyVersion()->getTableSchema(tableID);
+        auto property = tableSchema->getProperty(propertyID);
+        if (tableSchema->isNodeTable) {
+            WALReplayerUtils::initPropertyMetaDAsOnDisk(
```

Comment:
This looks a bit wrong to me. I was expecting you to go through the regular WAL version of pages mechanisms to checkpoint the necessary pages on disk. And then call something like a "checkpointInMemory()" function on NodeTable to create the NodeColumns (and in the InMemDiskArrays) that are being created as part of the add property DDL statement. I am even curious if we are already doing the WAL version of pages way of checkpointing as well as WALReplayerUtils::initPropertyMetaDAsOnDisk. It's not obvious to me that we are not doing such "double checkpointing" on disk.

Reply:
I'm separating this change into another PR.